src/study_groups.py

```
@@ -0,0 +1,670 @@
+        "SELECT id FROM study_group_invites"
+        " WHERE activity_id=? AND invitee_id=? AND status='pending'"
+    ).bind(group_id, invitee.id).first()
+    if pending:
+        return helpers["err"]("A pending invite already exists for this user", 400)
+
+    try:
+        await env.DB.prepare(
+            "INSERT INTO study_group_invites (id,activity_id,inviter_id,invitee_id,status)"
+            " VALUES (?,?,?,?,?)"
+        ).bind(helpers["new_id"](), group_id, user["id"], invitee.id, "pending").run()
+    except Exception as exc:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)